diesel_codegen_syntex/src/model.rs

```diff
@@ -25,13 +26,17 @@ impl Model {
         if let Annotatable::Item(ref item) = *annotatable {
             let table_name_from_annotation =
                 str_value_of_attr_with_name(cx, &item.attrs, "table_name");
+            let primary_key_name =
+                ident_value_of_attr_with_name(cx, &item.attrs, "primary_key")
+                    .unwrap_or(str_to_ident("id"));
```

Comment:
You're right on the syn side. libsyntax *definitely* doesn't have that impl though.